### ISSUE TITLE
no backward pass? 

### ISSUE THREAD

**Issue by etienne87** [itayhubara/BinaryNet.pytorch#6]
i'm printing the weights of the network and they are not changing. It makes sense since all the binarization is happening only on the _data_.

how can this code trains a networks for scratch with binarization?

**Comment by appleleaves** [comment]
In the file main_binary.py line 252, there is the loss.backward().
I think the backward pass has already done!
Also, are you really sure that the weights are not changing?
Maybe you should try a higher leaning rate to make sure the code is ok.
I think the gradient may be too small to change the signs of the weights.
@etienne87 @jafermarq @itayhubara
